ci/infra/testrunner/tests/test_dockercaps.py

```diff
@@ -50,29 +50,19 @@
       command: ['/bin/sh', '-c', 'sleep 3600']"""
 
 
-@pytest.fixture
-def setup_manifest():
-    cwd = pathlib.Path.cwd()
-    p = cwd / "pods.yaml"
-    p.touch()
-    p.write_text(textwrap.dedent(CONTENT))
-    yield p
-    p.unlink()
-
-
 @pytest.mark.flaky
-def test_dockercaps(deployment, kubectl, setup_manifest):
+def test_dockercaps(deployment, kubectl):
     logger.info("Deploy testcases")
     kubectl.run_kubectl(
-        "apply -f {}".format(setup_manifest))
+        "apply -f -", stdin=MANIFEST.encode())
 
-    wait(kubectl.run_kubectl,
-         "wait --for=condition=ready pods --all --timeout=0",
```

Comment:
I think the problem is that this is using `--timeout=0`. which according to the docs:

`--timeout=30s: The length of time to wait before giving up.  Zero means check once and don't wait, negative means
wait for a week.`

So it will only every check once and not really do a wait - I guess just passing any timeouts to the `kubectl wait` would already help, as then it would actually wait for the resources.

Reply:
Not sure I follow you. I'm removing the `wait` in favor of pulling the status and retrying using the wait function.